### PR TITLE
[AIRFLOW-3841] Remove DagBag from /tree

### PR DESCRIPTION
**Pull request by ffinfo**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title.
  - https://issues.apache.org/jira/browse/AIRFLOW-3841

### Description

Remove DagBag from /tree to make the webserver more stateless

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

File: airflow/www/views.py [hunk 1]
```diff
@@ -1185,16 +1186,16 @@ def tree(self, session=None):
...
-        dag_runs = (
-            session.query(DR)
-            .filter(
-                DR.dag_id == dag.dag_id,
-                DR.execution_date <= base_date)
-            .order_by(DR.execution_date.desc())
-            .limit(num_runs)
-            .all()
-        )
+        with create_session() as session:
```
Comment (ashb) [inline airflow/www/views.py:1188]: We've already got a `session` object in scope here.

Comment (ffinfo) [reply airflow/www/views.py:1188]: not anymore, just did remove that

File: airflow/models/__init__.py [hunk 2]
```diff
@@ -2574,19 +2574,21 @@ def clear(self,
 
         return count
 
-    def get_task_instances(self, session, start_date=None, end_date=None):
+    def get_task_instances(self, start_date=None, end_date=None):
```
Comment (ashb) [inline airflow/models/__init__.py]: Peter and I have been chatting about this, and what the right approach is.

I'd like us to move torward less granular commits, particularly in the webserver, following the flow form https://docs.sqlalchemy.org/en/latest/orm/contextual.html?highlight=contextual#using-thread-local-scope-with-web-applications which basically says "start a transaction at the start of a request, run everything in that one transaction, and commit once at the end".

This change takes us further from that.

I would vote for leaving this as is -- it _allows_  us to have control over what transaction this FN runs in, but by default (if it is not passed) it will run in it's own transaction anyway.

In short I don't think this code change is necessary: it removes control, and doesn't change the default behaviour

Comment (ffinfo) [reply airflow/models/__init__.py]: currently I did add back the session but now with provide_session. This keeps the possibility to reuse an existing session but does not make it required
